### PR TITLE
Sign up form

### PR DESCRIPTION
**Pull request by Oluwadamilareolusakin**
# Description
User sign up form

# What does it do?
It allows  a new user to be created in our database

# How does it work?
When you start the local server using `rails s`, access the app via localhost:3000 where you will find a
form that takes user details and saves them to the database

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  root 'users#new'
+  resources :users, only: %i[create new edit]
```
Comment (adriaanbd) [inline config/routes.rb]: Are these all the methods that you need? Did you compare these methods with the ones in `users_controller.rb`? Have you tried to edit an user by accessing `/users/1/edit`? Please create a new user and try it accessing the the page.

[`REQUIRED`]

File: app/views/users/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<%= form_for @user do |f| %>
```
Comment (adriaanbd) [inline app/views/users/_form.html.erb]: Have you tried accessing the edit path after creating an user `users/1/edit`? This is raising an error. Take a look at your `routes.rb` and see if if they're all right.

In addition, also check your routes with `$ rails routes` or more specifically `$rails routes | grep user`. Take note of the fact that there is no `user_path` available and that `@user` is converting to `user_path`.

We need to be able to edit an user.

[`REQUIRED`]

Comment (Oluwadamilareolusakin) [reply app/views/users/_form.html.erb]: Thank you for pointing that out, I will take a look at it.

Comment (Oluwadamilareolusakin) [reply app/views/users/_form.html.erb]: I just fixed the issue and pushed again

**Comment by Oluwadamilareolusakin** [comment]
Thanks for the review @adriaanbd

**Comment by Oluwadamilareolusakin** [comment]
@bellom Thanks a lot!

**Pull request merged**